integration-tests/miscellaneous/utilityBatch.test.ts

```diff
@@ -0,0 +1,45 @@
+import assert from "assert";
+import { KeyringPair } from "@polkadot/keyring/types";
+import { ExtrinsicHelper } from "../scaffolding/extrinsicHelpers";
+import { createAndFundKeypair } from "../scaffolding/helpers";
+import { ApiTypes, SubmittableExtrinsic } from "@polkadot/api/types";
+
+describe("Utility Batch Filtering", function () {
+    let sender: KeyringPair;
+    let recipient: KeyringPair;
+
+    before(async function () {
+        sender = await createAndFundKeypair();
+        recipient = await createAndFundKeypair();
+    });
+
+    it("should successfully execute ✅ batch with allowed calls", async function () {
+        // good batch: with only allowed calls
+        const goodBatch: SubmittableExtrinsic<ApiTypes>[] = [];
+        goodBatch.push(ExtrinsicHelper.api.tx.balances.transfer(recipient.address, 1000))
+        goodBatch.push(ExtrinsicHelper.api.tx.system.remark("Hello From Batch"))
+        goodBatch.push(ExtrinsicHelper.api.tx.msa.create())
+        const batch = ExtrinsicHelper.executeUtilityBatchAll(sender, goodBatch);
+        const [event, eventMap] = await batch.fundAndSend();
+        assert.notEqual(event, undefined, "should return an event");
+        assert.notEqual(eventMap, undefined, "should return an eventMap");
+    });
+
+    it("should fail to execute ❌ batch with disallowed calls", async function () {
```

Comment:
This would fail as long as any one of the calls inside the batch is bad. I think you'd need to look over them with "one" call per batch attempt

Reply:
Added more tests to cover some cases